RecoTracker/FinalTrackSelectors/plugins/DuplicateListMerger.cc

```diff
@@ -205,309 +163,121 @@ void DuplicateListMerger::produce(edm::Event& iEvent, const edm::EventSetup& iSe
   MVACollection mvaVec;
 
 
-  if(copyExtras_){
-    outputTrkExtras = std::auto_ptr<reco::TrackExtraCollection>(new reco::TrackExtraCollection);
-    outputTrkExtras->reserve(originalHandle->size());
-    refTrkExtras    = iEvent.getRefBeforePut<reco::TrackExtraCollection>();
-    outputTrkHits   = std::auto_ptr<TrackingRecHitCollection>(new TrackingRecHitCollection);
-    outputTrkHits->reserve(originalHandle->size()*25);
-    refTrkHits      = iEvent.getRefBeforePut<TrackingRecHitCollection>();
-    if (makeReKeyedSeeds_){
-      outputSeeds = std::auto_ptr<TrajectorySeedCollection>(new TrajectorySeedCollection);
-      outputSeeds->reserve(originalHandle->size());
-      refTrajSeeds = iEvent.getRefBeforePut<TrajectorySeedCollection>();
-    }
-  }
+  auto mergedMVA = *mergedMVAStore;
 
   //match new tracks to their candidates
-  std::vector<std::pair<int,const DuplicateRecord*> > matches;
-  for(int i = 0; i < (int)candidateHandle->size(); i++){
-    const DuplicateRecord& duplicateRecord = candidateHandle->at(i);
-    int matchTrack = matchCandidateToTrack(duplicateRecord.first,mergedHandle);
-    if(matchTrack < 0)continue;
-    const reco::Track& matchedTrack(mergedTracks[matchTrack]);
-    if( ChiSquaredProbability(matchedTrack.chi2(),matchedTrack.ndof()) < minTrkProbCut_)continue;
-    unsigned int dHits = (duplicateRecord.first.recHits().second - duplicateRecord.first.recHits().first) - matchedTrack.recHitsSize();
+  std::vector<std::array<int,3>> matches;
+  for(int i = 0; i < (int)merged.size(); ++i) {
+    auto cInd = candIndices[i];
+    auto const & cand = candidates[cInd];
+    const reco::Track& matchedTrack = merged[i];
+
+    if (mergedMVA[i]< -0.7f) continue;  // at least "loose"
```

Comment:
looks like a temporary magic number, 
esp. considering no strictly defined range for an MVA or even no particularly deep meaning of -0.7 even if the range remains fixed: a retraining can give some other optimal value.